sw/device/lib/crypto/impl/rsa/rsa_padding.c

```diff
@@ -32,6 +43,27 @@ static const uint8_t kSha512DigestIdentifier[] = {
     0x40, 0x04, 0x00, 0x05, 0x03, 0x02, 0x04, 0x03, 0x65, 0x01,
     0x48, 0x86, 0x60, 0x09, 0x06, 0x0d, 0x30, 0x51, 0x30,
 };
+/*
+ * SHA-3 digest identifiers adapted from the SHA-2 identifers based on the
+ * algorithm identifiers on
+ * https://csrc.nist.gov/projects/computer-security-objects-register/algorithm-registration
+ */
+static const uint8_t kSha3_224DigestIdentifier[] = {
```

Comment:
Only a few bytes change between each of these identifiers:

- Byte 0: Length of the hash digest
- Byte 4: Last number of the OID for the hash function (as defined in https://csrc.nist.gov/projects/computer-security-objects-register/algorithm-registration)
- Byte 17: Length of the entire DER sequence containing the OID and message digest, excluding the final `0x30` byte, which is the opcode for the DER sequence itself.

Reply:
Just to make sure the encoding is right -- have you checked this against an externally generated test vector yet?

Reply:
I've run the wycheproof test vectors through this code. I'm getting a couple sporadic test failures that aren't unique to the SHA-3 hashes. We'll have to talk through these at some point this week, but they shouldn't block this PR.